pkg/pgmodel/querier/query_builder.go

```diff
@@ -279,6 +279,21 @@ func (c *clauseBuilder) Build(includeMetricName bool) ([]string, []interface{},
 
 func buildTimeSeries(rows []timescaleRow, lr lreader.LabelsReader) ([]*prompb.TimeSeries, error) {
 	results := make([]*prompb.TimeSeries, 0, len(rows))
+	labelIDMap := make(map[int64]labels.Label)
+	for i := range rows {
+		for _, id := range rows[i].labelIds {
+			//id==0 means there is no label for the key, so nothing to look up
+			if id == 0 {
+				continue
+			}
+			labelIDMap[id] = labels.Label{}
+		}
+	}
+
+	err := lr.LabelsForIdMap(labelIDMap)
+	if err != nil {
+		return nil, err
```

Comment:
```suggestion
		return nil, fmt.Errorf("build timeseries: %w ", err)
```

Lets wrap the error here, otherwise we can confuse with the error at line 300.